core/slice.go

```diff
@@ -516,13 +514,22 @@ func (sl *Slice) readPhCache(hash common.Hash) (types.PendingHeader, bool) {
 				return types.PendingHeader{}, false
 			}
 		}
+	} else {
+		ph := rawdb.ReadPendingHeader(sl.sliceDb, hash)
+		if ph != nil {
+			sl.phCache.Add(hash, ph)
+			return *types.CopyPendingHeader(ph), exists
```

Comment:
`exists` returns `false` here but should return `true`

Reply:
good catch